hardware_interface/include/hardware_interface/system.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__SYSTEM_HPP_
+#define HARDWARE_INTERFACE__SYSTEM_HPP_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/component_interfaces/system_interface.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+class System final
```

Comment:
Before merging, check compatibility with interface and if all member functions are exposed.

Reply:
This will be discussed in ros-controls/ros2_control#101